core/chains/evm/gas/block_history_estimator.go

```diff
@@ -223,7 +225,13 @@ func (b *BlockHistoryEstimator) GetLegacyGas(_ []byte, gasLimit uint32, maxGasPr
 		return nil, 0, errors.New("BlockHistoryEstimator is not started; cannot estimate gas")
 	}
 	if gasPrice == nil {
-		return nil, 0, errors.Errorf("GasPrice unset. Likely cause is that BlockHistoryEstimator failed to make the first estimation on start, or no suitable transactions are available for estimation (size=%d)", b.size)
+		if !b.initialFetch.Load() {
+			return nil, 0, errors.New("BlockHistoryEstimator has not finished the first gas estimation yet, likely because a failure on start")
+		}
+		b.logger.Warnw(fmt.Sprintf("BlockHistoryEstimator couldn't provide an estimation. This is likely because"+
+			"there aren't any valid transactions to estimate from. Using EvmDefaultGasPriceWei as fallback."+
+			"(block numbers: %v .", b.printBlockHistoryNumbers()...))
```

Comment:
I think it would be easier to read if the func just returned the `numsInHistory []int64`, plus it is more type safe.
```suggestion
			"(block numbers: %v .", "blocks", b.printBlockHistoryNumbers()))
```

Reply:
Oh, and actually are these meant to be passed to the logger? Not as Sprintf args?

Reply:
How about this?
```suggestion
						"blocks", b.getBlockHistoryNumbers())
						
						func (b *BlockHistoryEstimator) getBlockHistoryNumbers() (numsInHistory []int64) {
	                                             for _, b := range b.blocks {
		                                         numsInHistory = append(numsInHistory, b.Number)
	                                             }
	                                             return
                                                 }
```